### PR TITLE
[Improve](inverted_index) update clucene and improve array inverted index writer 

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/inverted_index_writer.cpp [hunk 1]
```diff
@@ -369,39 +388,45 @@ class InvertedIndexColumnWriterImpl : public InvertedIndexColumnWriter {
...
+                         v->get_size() > ignore_above) ||
+                        (_parser_type != InvertedIndexParserType::PARSER_NONE && v->empty())) {
+                        // is here a null value?
+                        RETURN_IF_ERROR(add_null_document());
+                    } else {
+                        new_fulltext_field(v->get_data(), v->get_size(), new_field);
+                        _doc->add(*new_field);
+                    }
                 }
+                RETURN_IF_ERROR(add_document());
```
Comment (airborne12) [inline be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:428]: if already add_null_document, then add_document() will cause unexpected problem

Comment (amorynan) [reply be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:428]: so maybe we should not make this if branch?

**Comment by amorynan** [comment]
run buildall

**Review by airborne12 — APPROVED** [review]
LGTM

**Review by zzzxl1993 — APPROVED** [review]
LGTM

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run p0

**Comment by amorynan** [comment]
run p0

**Comment by amorynan** [comment]
run external

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run p0

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Review by airborne12 — APPROVED** [review]
LGTM

File: be/src/olap/rowset/segment_v2/inverted_index_writer.cpp [hunk 2]
```diff
@@ -348,39 +367,60 @@ class InvertedIndexColumnWriterImpl : public InvertedIndexColumnWriter {
...
-                    new_fulltext_field(value.c_str(), value.length());
-                    RETURN_IF_ERROR(add_document());
+                    if ((_parser_type == InvertedIndexParserType::PARSER_NONE &&
+                         v->get_size() > ignore_above) ||
+                        (_parser_type != InvertedIndexParserType::PARSER_NONE && v->empty())) {
+                        // is here a null value?
+                        // TODO. Maybe here has performance problem for large size string.
+                        continue;
+                    } else {
+                        if (_parser_type == InvertedIndexParserType::PARSER_ENGLISH ||
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:405]: why check specific parser type? We should check NONE and others.

File: be/src/olap/rowset/segment_v2/inverted_index_writer.cpp [hunk 3]
```diff
@@ -293,6 +301,17 @@ class InvertedIndexColumnWriterImpl : public InvertedIndexColumnWriter {
         }
     }
 
+    void new_fulltext_field(const char* field_value_data, size_t field_value_size,
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:304]: this function is not used

Comment (amorynan) [reply be/src/olap/rowset/segment_v2/inverted_index_writer.cpp]: done!

**Comment by amorynan** [comment]
run buildall

**Review by airborne12 — APPROVED** [review]

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**